src/pal/pal_posix.h

```diff
@@ -342,6 +349,30 @@ namespace snmalloc
         if (getentropy(&result, sizeof(result)) != 0)
           error("Failed to get system randomness");
         return result;
+#elif defined(SYS_getrandom)
+        union
+        {
+          uint64_t result;
+          char buffer[sizeof(uint64_t)];
+        };
+        int cs;
+        auto current = std::begin(buffer);
+        auto target = std::end(buffer);
+        pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, &cs);
+        while (current != target)
+        {
+          auto ret = syscall(SYS_getrandom, current, target - current, 0);
+          if (ret < 0)
+          {
+            if (errno == EINTR)
```

Comment:
Could this cause an infinite loop?  e.g. if we get an `ENOSYS`.  Should we hard error on that as it is not the expected behaviour?